src/org/aavso/tools/vstar/plugin/ob/sink/impl/TextFormatObservationSinkPlugin.java

```diff
@@ -1,142 +0,0 @@
-/**
- * VStar: a statistical analysis tool for variable star data.
- * Copyright (C) 2010  AAVSO (http://www.aavso.org/)
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Affero General Public License as
- * published by the Free Software Foundation, either version 3 of the
- * License, or (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Affero General Public License for more details.
- *
- * You should have received a copy of the GNU Affero General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
- */
-package org.aavso.tools.vstar.plugin.ob.sink.impl;
-
-import java.io.IOException;
-import java.io.PrintWriter;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-
-import org.aavso.tools.vstar.data.ValidObservation;
-import org.aavso.tools.vstar.plugin.ObservationSinkPluginBase;
-import org.aavso.tools.vstar.ui.mediator.AnalysisType;
-import org.aavso.tools.vstar.ui.mediator.Mediator;
-import org.aavso.tools.vstar.ui.mediator.NewStarType;
-import org.aavso.tools.vstar.ui.mediator.ViewModeType;
-import org.aavso.tools.vstar.ui.mediator.message.ProgressInfo;
-import org.aavso.tools.vstar.util.locale.LocaleProps;
-
-/**
- * An observation sink plugin for simple text and AAVSO Download format file
- * saves.
- */
-public class TextFormatObservationSinkPlugin extends ObservationSinkPluginBase {
-
-	private final static Map<String, String> DELIMS;
-
-	private final static Map<String, String> SUFFIXES;
-
-	static {
-		DELIMS = new TreeMap<String, String>();
-		DELIMS.put("Comma", ",");
-		DELIMS.put("Space", " ");
-		DELIMS.put("Tab", "\t");
-		
-		SUFFIXES = new TreeMap<String, String>();
-		SUFFIXES.put("Comma", "csv");
-		SUFFIXES.put("Space", "txt");
-		SUFFIXES.put("Tab", "tsv");
-	}
-
-	@Override
-	public void save(PrintWriter writer, List<ValidObservation> obs,
-			String delimiter) throws IOException {
-
-		if (Mediator.getInstance().getLatestNewStarMessage().getNewStarType() == NewStarType.NEW_STAR_FROM_SIMPLE_FILE) {
```

Comment:
Was the issue that this check wasn't behaving as expected when saving residuals? I.e. it thought it should be saved in simple format when it shouldn't since it wasn't a new star?

Reply:
Hi @mdadighat. Yes, that's pretty much it.

Figuring out why what's changed, has changed, based upon the diffs is probably quite confusing here.

This check on line 61 above, that you show, of the initial implementation of the text format observation sink plugin made an incorrect assumption about what format to save the file as, based upon the latest dataset loaded.

I honestly look at that initial plugin code now and I wonder what I was thinking.

Before the addition of the observation sink plugin interface, the Mediator code passed the so-called "new star type" (simple format, AAVSO download format, arbitrary format) as a parameter to the ObservationSaveTask. After the plugin interface was added, the plugin tried to determine this "new star type" on its own, incorrectly.

The plugin initially appeared as a unified "Simple or Download" format item in the save dialog, whereas now, there are two plugins, for simple and download. Now there is no ambiguity. Either we are saving as simple format or download format, explicitly.

I hope that makes sense.

Reply:
Yep, makes sense! Splitting the plugin into two seemed straight forward but I just wanted to double check that I understood the specific reason for it. I wonder if this was the only case affected or just the only one that got noticed.

Reply:
This was the only case because the only two save mechanisms before the obs sink plugin mechanism was added.